contracts/Market.sol

```diff
@@ -389,26 +389,40 @@ contract Market is IMarket, Ownable, ERC721 {
 		return _bets[index].payout;
 	}
 
-
 	function _payout(uint256 index, uint8 result) internal virtual {
-		if (result == SCRATCHED) {
-			uint256 lay = _bets[index].payout - _bets[index].amount;
+		uint256 payout = _bets[index].payout;
+		uint256 amount = _bets[index].amount;
+		uint256 loan = payout - amount;
+
+		if (result == SCRATCHED) {				
 			// Transfer the bet amount to the owner of the NFT
-			IERC20(_vault.asset()).transfer(ownerOf(index), _bets[index].amount);
-			// Transfer the lay back to the vault
-			IERC20(_vault.asset()).transfer(address(_vault), lay);
-		} else {
-			address recipient = result == WINNER ? ownerOf(index) : address(_vault);
-			uint256 amount = _bets[index].payout;
+			IERC20(_vault.asset()).transfer(ownerOf(index), amount);
```

Comment:
not sure how it affects gas much _but_ why not just assign this to a variable and call it, looks like its being reinitialized a lot:
```
IERC20 vaultAsset = IERC20(_vault.asset());
```

Reply:
Yes, this would reduce OpEx gas as each of these are an external call and an SSLOAD